traitlets/traitlets.py

```diff
@@ -1644,6 +1644,17 @@ def trait_events(cls, name=None):
                         events[k] = v
         return events
 
+    def clear(self, **metadata):
+        """Remove the values of traits with certain metadata.
+
+        If not metadata is given, then all trait values are cleared.
+        """
+        for name in self.traits(**metadata):
+            try:
+                self._trait_values[name]
```

Comment:
Should this be `del self._trait_values[name]`?

Reply:
Yes, thank you.

Reply:
This also needs a test.